src/transport/raw/TCP.h

```diff
@@ -104,15 +104,7 @@ class DLL_EXPORT TCPBase : public Base
             size_t packetsBuffersSize) :
         mActiveConnections(activeConnectionsBuffer),
         mActiveConnectionsSize(bufferSize), mPendingPackets(packetBuffers), mPendingPacketsSize(packetsBuffersSize)
-    {
-        PendingPacket emptyPending{
-            .peerAddress  = PeerAddress::Uninitialized(),
-            .packetBuffer = nullptr,
-        };
-
-        std::fill(mActiveConnections, mActiveConnections + mActiveConnectionsSize, nullptr);
-        std::fill(mPendingPackets, mPendingPackets + mPendingPacketsSize, emptyPending);
```

Comment:
So this is relying on the default ctor of `PeerAddress` to produce the same thing as `PeerAddress::Uninitialized()`, right?  And relying on the caller giving us an array that hasn't been used before?  Can we actually rely on that?  If we can, please document the assumptions in the API documentation.

Reply:
This is _currently_ only called from one place (`TCP`) such that the fills match default construction, but I'll make them explicit again in case that changes. Maybe I was just a little annoyed that `std::fill` doesn't handle move construction (yet).

Reply:
OK, that wasn't right…

Reply:
Fun.

I think we could probably get away with documenting that the caller needs to ensure the fields are initialized properly and skip the extra work here, since it's going to happen in the subclass ctor anyway in practice...  Doing that would be fine with me.  Or a followup to decide what to do.

Reply:
For now I'll leave it as it is now, with an explicit loop and comment; at least no one should get bitten by it again.